### PR TITLE
Bug 1115773 - Allow L10n's loadJSON to be sync. r=gandalf

### PR DESCRIPTION
**Pull request by stasm**
https://bugzilla.mozilla.org/show_bug.cgi?id=1115773

### CODE REVIEW

**Comment by try-server-hook** [comment]
[stasm Staś Małolepszy (stasm)](https://github.com/stasm) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=2e3f92a892d8)

**Comment by try-server-hook** [comment]
[stasm Staś Małolepszy (stasm)](https://github.com/stasm) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=c21afc56e8ee)

File: shared/js/l10n.js [hunk 1]
```diff
@@ -15,46 +15,33 @@
...
-      xhr.open('GET', url);
-
-      xhr.responseType = 'json';
-      xhr.addEventListener('load', function io_loadjson(e) {
+      xhr.addEventListener('load', function io_onload(e) {
         if (e.target.status === 200 || e.target.status === 0) {
-          callback(null, e.target.response);
+          callback(
+            null, needParse ?
+              JSON.parse(e.target.response) : e.target.response);
```
Comment (stasm) [inline shared/js/l10n.js]: This uses the fact that by default, `e.target.response` is of type 'string'.

Comment (stasm) [inline shared/js/l10n.js]: It turns out Sinon'JS's FakeXHR doesn't expose the `response` property, so I had to go back to falling back onto `responseText` for tests.

**Comment by try-server-hook** [comment]
[stasm Staś Małolepszy (stasm)](https://github.com/stasm) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=f856cc98486e)

**Comment by try-server-hook** [comment]
[stasm Staś Małolepszy (stasm)](https://github.com/stasm) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=917318717156)

**Pull request merged**